source/v0/TCPStream.cpp

```diff
@@ -68,3 +68,14 @@ void TCPStream::_eventHandler(struct socket_event *ev)
             break;
     }
 }
+
+void TCPStream::setNagle(bool enable)
+{
+    void * enable_ptr;
+    if(enable) {
+        enable_ptr = (void*)1;
```

Comment:
It's just because ```void *``` is inherently unsafe to start with. It simply encodes 32-bits of arbitrary information. Typically, those 32 bits are used to encode a pointer, but they don't have to. I only needed one bit.

The other point is that, should another stack have configuration parameters to the nagle implementation (for example artificially lowering the maximum segment size for which you want nagle to operate), you can use the pointer to pass a structure with other config values. In this case, a NULL pointer indicates disable, while a non-NULL pointer with zero size indicates enable, and a non-NULL pointer with non-zero size indicates enable with config.